crates/flycheck/src/lib.rs

```diff
@@ -339,12 +339,7 @@ impl CargoActor {
                         cargo_metadata::Message::CompilerMessage(msg) => {
                             self.sender.send(CargoMessage::Diagnostic(msg.message)).unwrap()
                         }
-
-                        cargo_metadata::Message::CompilerArtifact(_)
-                        | cargo_metadata::Message::BuildScriptExecuted(_)
-                        | cargo_metadata::Message::BuildFinished(_)
-                        | cargo_metadata::Message::TextLine(_)
-                        | _ => (),
+                        _ => (),
```

Comment:
Ugh

Reply:
?

Reply:
It's fine, it makes sense, but the default branch was a bit weird (we might want to know if new message types are added).

Reply:
Well we won't know that though because due to non exhaustiveness we have to use a wildcard no matter what

Reply:
Yeah, it's was an "ugh" for the old code, not for your change.

Reply:
Upstream added non-exhaustive, that's why we need this here. Last weekend I almost got to prototyping miniserde-based cargo data to have more control over this :-)